backend/src/test/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiServiceTest.java

```diff
@@ -0,0 +1,93 @@
+        String artistImageUrl = "https://img.discogs.com/wcD5QViPvggOaC3T_D5ql1rHY_E=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-12596-1565996731-5432.jpeg.jpg";
+        String discogsArtistUrl = "https://api.discogs.com/artists/6666";
+        String baseUrl = "https://api.discogs.com";
+        String discogsApiUrl = baseUrl + "/database/search?type=artist&q=" + artistName;
+        String accessToken = "testtest";
+
+        DiscogsArtist testDiscogsArtist = DiscogsArtist.builder()
+                .discogsArtistId(discogsArtistId)
+                .artistName(artistName)
+                .artistImageUrl(artistImageUrl)
+                .discogsArtistUrl(discogsArtistUrl).build();
+
+        DiscogsArtistSearchResults testDiscogsArtistSearchResults = DiscogsArtistSearchResults.builder()
+                .results(new DiscogsArtist[]{testDiscogsArtist}).build();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("User-Agent", "MusiCompass/0.1");
+        headers.add("Authorization", "Discogs token=" + accessToken);
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
+
+        ResponseEntity<DiscogsArtistSearchResults> mockResponseEntity = ResponseEntity.ok(testDiscogsArtistSearchResults);
+
+        when(restTemplate.exchange(discogsApiUrl, HttpMethod.GET, entity, DiscogsArtistSearchResults.class))
+                .thenReturn(mockResponseEntity);
+
+        //WHEN
+        DiscogsArtistSearchResults actual = discogsArtistApiService.getDiscogsArtistByArtistName(artistName, accessToken);
+
+        //THEN
+        assertThat(actual, equalTo(testDiscogsArtistSearchResults));
+    }
+
+    @Test
+    @DisplayName("Get empty return after trying to find non-existing artist")
+    public void returnsExceptionTryingToFindNonExistingArtist(){
+        //GIVEN
+        String artistName = "bliblablubb";
+        String baseUrl = "https://api.discogs.com";
+        String discogsApiUrl = baseUrl + "/database/search?type=artist&q=" + artistName;
+        String accessToken = "testtest";
+
+        DiscogsArtistSearchResults testDiscogsArtistSearchResults = DiscogsArtistSearchResults.builder()
+                .results(new DiscogsArtist[]{}).build();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("User-Agent", "MusiCompass/0.1");
+        headers.add("Authorization", "Discogs token=" + accessToken);
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
+
+        ResponseEntity<DiscogsArtistSearchResults> mockResponseEntity = ResponseEntity.ok(testDiscogsArtistSearchResults);
+
+        when(restTemplate.exchange(discogsApiUrl, HttpMethod.GET, entity, DiscogsArtistSearchResults.class))
+                .thenReturn(mockResponseEntity);
+
+        //WHEN
+        DiscogsArtistSearchResults actual = discogsArtistApiService.getDiscogsArtistByArtistName(artistName, accessToken);
+
+        //THEN
+        assertThat(actual.getResults(), emptyArray());
```

Comment:
is this assertThat working? you are missing an operation in your second input e.g. `is()` or `equals()` which I would expect in hemcrest `assertThat`
if checking an empty array you can also use `.length()` and ensure it matches e.g. `is(0)`

Reply:
it is working this way...